### ISSUE TITLE
Prevent Timing Attacks

### ISSUE THREAD

**Issue by zimt28** [BlakeWilliams/doorman#12]
I'm currently reading _Programming Phoenix_ and right there the authentication method is defined like this:

``` elixir
def login_by_username_and_pass(conn, username, given_pass, opts) do
  repo = Keyword.fetch!(opts, :repo)
  user = repo.get_by(Rumbl.User, username: username)

  cond do
    user && checkpw(given_pass, user.password_hash) ->
      {:ok, login(conn, user)}
    user ->
      {:error, :unauthorized, conn}
    true ->
      dummy_checkpw()
      {:error, :not_found, conn}
  end
end
```

Quite similar to [doorman's](https://github.com/BlakeWilliams/doorman/blob/master/lib/doorman.ex#L18-L25), but it also uses `Comeonin`'s `dummy_checkpw` function to prevent [timing attacks](https://en.wikipedia.org/wiki/Timing_attack). It would be great if you could add this as well :)

**Comment by BlakeWilliams** [comment]
Good call! I'll look into implementing this.

**Comment by BlakeWilliams** [comment]
Thanks for reporting this, fixed in #13.
